### ISSUE TITLE
`ipfs add` does not support --enc=json

### ISSUE THREAD

**Issue by natewalck** [ipfs/go-ipfs#4459]
ipfs version --all
go-ipfs version: 0.4.13-
Repo version: 6
System version: amd64/linux
Golang version: go1.9.2

#### Version information:

Enhancement? Bug?

#### Type:
Low

#### Severity:
It seems like the command should support --enc=json much like `ipfs stats` supports it.

#### Description:
When running `ipfs add --enc=json` it does not return the output in json.

```
ipfs add -rw somefolder --enc=json
added QmTKzuL8FNd1cc9Zu1qP1a9PbgJYg5pN5vBB92Q6St3Soc somefolder/folder/path
...
```

However, it does work when running`ipfs stats bw --enc=json` or any other `ipfs stats` commands:

```
ipfs stats bw --enc=json
{"TotalIn":351838,"TotalOut":352951,"RateIn":0.0000037126491990256317,"RateOut":0.0000037126491990256317}
```

I'm not sure if this is a bug or an enhancement request (I'd expect --enc=json to work everywhere, but obviously there are good reasons why this might not be the case).

**Comment by keks** [comment]
Thanks for filing this. Your expectations are correct, --enc=json should work everywhere. It's just that there was a big refactor in how commands are passed and I introduced this regression. Oh well :)

**Comment by Kubuxu** [comment]
@keks it was like that before.

**Comment by keks** [comment]
Huh, I'm glad it's not my fault, then 😁. I'm pretty busy at the moment, so if we plan to get this fixed for 0.4.14 maybe someone else can take over? I assigned this to me because I thought it was due to the refactor.

**Comment by frrist** [comment]
There was previous discussion on this issue here - #1121. Closing in favor of it.
